src/main/java/com/aws/greengrass/util/platforms/unix/linux/LinuxSystemResourceController.java

```diff
@@ -63,30 +63,36 @@ public void updateResourceLimits(GreengrassService component, Map<String, Object
             }
             if (resourceLimit.containsKey(MEMORY_KEY)) {
                 long memoryLimitInKB = Coerce.toLong(resourceLimit.get(MEMORY_KEY));
-
-                // TODO: add input validation
-                String memoryLimit = Long.toString(memoryLimitInKB * 1024);
-                Files.write(Cgroup.Memory.getComponentMemoryLimitPath(component.getServiceName()),
-                        memoryLimit.getBytes(StandardCharsets.UTF_8));
+                if (memoryLimitInKB > 0) {
```

Comment:
Is it possible to do this validation earlier? In this case if the 0 value is due to missing input from customer but there's a default present, that default won't be applied right?

Reply:
I looked through the code and couldn't find a better place to do the validation. If the whole `SystemResourceLimits` is missing, then the default value under `Nuclues` component is used. However, if only the `cpu` or `memory` field is missing, then this validation logic will kick in.